### PR TITLE
[6.2.x] Backport Python 3.10 support (PRs #8494 and #8555)

### LINKED ISSUE #8638: Please backport Python 3.10 fixes to 6.2.x
Unless I'm mistaken, the 6.2.4 release seems to be entirely fine with Python 3.10, except for a bunch of test failures. The failures seem to be specifically related to tests themselves, and I've been able to get them to pass by cherry-picking the following commits:

913cffa45f9230087f589140604ec1eb5f199a83 f1570dc038585c8e5746f9bfdeb76321cd8be722 710446420c0bb98f0761724ec865a894820bb610 382599287f4cab914bd7502887072e7f325cdb09

Given that these commits do not affect anything but tests, could you please consider backporting them to 6.2.x branch?

### PR DESCRIPTION
**Pull request by mgorny**
Fixes #8638

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -17,11 +17,11 @@ classifiers =
     Operating System :: POSIX
     Programming Language :: Python :: 3
     Programming Language :: Python :: 3 :: Only
+    Programming Language :: Python :: 3.10
     Programming Language :: Python :: 3.6
     Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
-    Programming Language :: Python :: 3.10
```
Comment (Zac-HD) [inline setup.cfg]: Updating [`setup-cfg-fmt` to version 1.17.0](https://github.com/pytest-dev/pytest/blob/6.2.x/.pre-commit-config.yaml#L42) in our pre-commit config (as on main) will fix this.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,8 +42,7 @@ repos:
     rev: v1.11.0
```
Comment (Zac-HD) [inline .pre-commit-config.yaml:42]:
```suggestion
    rev: v1.17.0
```

This is the version we're using on `main`, and it ensures that the `3.10` classifier sorts in version order (i.e. after `3.9`).

Comment (mgorny) [reply .pre-commit-config.yaml]: Done, thanks!

**Comment by nicoddemus** [comment]
Thanks @mgorny!

**Pull request merged**